### PR TITLE
Drop EEG rejection thresholds when replacing EEG with CSD channels

### PR DESCRIPTION
**Pull request by hoechenberger**
This is an attempt to fix the issue reported at https://mne.discourse.group/t/saving-epoch-error-no-eeg-channel-found

WIP:
- [x] tests
- [x] changelog
- [ ] update docstring to explain we're removing rejection thresholds for EEG

### CODE REVIEW

**Comment by hoechenberger** [comment]
If anyone's willing to take this one over, please feel free to do so as I'm very much busy with other stuff

cc @larsoner @adam2392 @sappelhoff @alexrockhill

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -92,6 +92,8 @@ Bugs
 
 - Fix bug in the :class:`mne.viz.Brain` tool bar that prevented the buttons to call the corresponding feature (:gh:`10560` by `Guillaume Favelier`_)
 
+- Fix issue with saving epochs once :func:`mne.preprocessing.compute_current_source_density` has been used if a rejection threshold was used first (:gh:`10619` by `Alex Rockhill`_ and `Richard Höchenberger`_)
```
Comment (hoechenberger) [inline doc/changes/latest.inc:95]:
```suggestion
- Fix issue with saving epochs once :func:`~mne.preprocessing.compute_current_source_density` has been used if a rejection threshold was used first (:gh:`10619` by `Alex Rockhill`_ and `Richard Höchenberger`_)
```

**Comment by alexrockhill** [comment]
@hoechenberger, I didn't update the docstring, I'm really not sure what it would say, I don't think this is a user-facing change. I think the behavior is now what's expected as in the epochs can be saved and loaded from a disk with a rejection threshold applied. You could add something to explain about that your rejection threshold for `eeg` would be lost, but I don't think a user would expect that data to still be there since the data is no longer in that form anyway and it's not really of any use since the units changed...

File: doc/changes/latest.inc (hunk 1 above)
Comment (alexrockhill) [reply doc/changes/latest.inc:95]: What does this do?

**Comment by hoechenberger** [comment]
Great work, @alexrockhill! And I agree, I think we don't need a docstring update after all. Feel free to merge once CI goes green!

File: doc/changes/latest.inc (hunk 1 above)
Comment (hoechenberger) [reply doc/changes/latest.inc:95]: only prints "compute_current_source_density" while still linking to "mne.preprocessing.compute_current_source_density"

Comment (alexrockhill) [reply doc/changes/latest.inc]: Nice

**Comment by alexrockhill** [comment]
Is there a fix to the pandas install coming or are we merging without those tests?

**Comment by hoechenberger** [comment]
I'd leave this decision to @larsoner 
As I'm going to bed now anyway 😁😴

**Comment by alexrockhill** [comment]
Thanks @hoechenberger!

**Comment by hoechenberger** [comment]
Thanks for bringing this home!

File: mne/preprocessing/_csd.py [hunk 2]
```diff
@@ -176,6 +176,14 @@ def compute_current_source_density(inst, sphere='auto', lambda2=1e-5,
     for pick in picks:
         inst.info['chs'][pick].update(coil_type=FIFF.FIFFV_COIL_EEG_CSD,
                                       unit=FIFF.FIFF_UNIT_V_M2)
+
+    # Remove rejection thresholds for EEG
+    if isinstance(inst, BaseEpochs):
+        inst.reject = None if inst.reject is None else \
+            {k: v for k, v in inst.reject.items() if k != 'eeg'}
```
Comment (agramfort) [inline mne/preprocessing/_csd.py:183]:
```
if inst.reject and "eeg" in inst.reject:
   del inst.reject["eeg"]
```
would be more readable.

Comment (hoechenberger) [reply mne/preprocessing/_csd.py:183]: Good idea!
I'll push a "fix"

**Pull request merged**